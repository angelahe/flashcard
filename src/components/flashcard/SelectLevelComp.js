import React from 'react';
import PropTypes from 'prop-types';
import '../../styles/flashcard.css';
import write from '../../img/buttons/write_FFFFFF.png';
import eye from '../../img/buttons/eye_FFFFFF.png';
import read from '../../img/buttons/read_FFFFFF.png';

const SelectLevelComp = (props) => {
  const {
    onLevelSelect,
  } = props;

  function handleReview() {
    onLevelSelect('Review');
  }

  function handleRecognize() {
    onLevelSelect('Recognize');
  }

  function handleProduce() {
    onLevelSelect('Produce');
  }

  return (
    <div className="ChooseLevelComp">
      <br /><br /><br />
      <div className="LineContainer">
        <button className="ReviewBtn" type="button" onClick={handleReview}>
          <img className="LevelImg" src={eye} alt="review" />
        </button>
      </div>
      <div className="LineContainer">
        <button className="RecognizeBtn" type="button" onClick={handleRecognize}>
          <img className="LevelImg" src={read} alt="recognize" />
        </button>
      </div>
      <div className="LineContainer">
        <button className="ProduceBtn" type="button" onClick={handleProduce}>
          <img className="LevelImg" src={write} alt="produce" />
        </button>
      </div>
    </div>
  );
};

SelectLevelComp.defaultProps = {
  onLevelSelect: () => { },
};

SelectLevelComp.propTypes = {
  onLevelSelect: PropTypes.func,
};

export default SelectLevelComp;
