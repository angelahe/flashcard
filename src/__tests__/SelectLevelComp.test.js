import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SelectLevelComp from '../components/flashcard/SelectLevelComp';

it('renders without crashing', () => {
  shallow(<SelectLevelComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SelectLevelComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  const tree = renderer.create(<SelectLevelComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div ChooseLevelComp on shallow render', () => {
  const component = shallow(<SelectLevelComp />);
  expect(component.html()).toContain('ChooseLevelComp');
});

it('calls onLevelPicked on Review click', (done) => {
  const handleReviewClick = jest.fn(() => {});
  const component = shallow(<SelectLevelComp
    onLevelSelect={handleReviewClick}
  />);
  const button = component.find('.ReviewBtn').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleReviewClick.mock.calls.length).toBe(1);
    expect(handleReviewClick.mock.calls[0][0]).toBe('Review');
    done();
  });
});

it('calls onLevelPicked on Recognize click', (done) => {
  const handleRecognizeClick = jest.fn(() => {});
  const component = shallow(<SelectLevelComp
    onLevelSelect={handleRecognizeClick}
  />);
  const button = component.find('.RecognizeBtn').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleRecognizeClick.mock.calls.length).toBe(1);
    expect(handleRecognizeClick.mock.calls[0][0]).toBe('Recognize');
    done();
  });
});

it('calls onLevelPicked on Produce click', (done) => {
  const handleProduceClick = jest.fn(() => {});
  const component = shallow(<SelectLevelComp
    onLevelSelect={handleProduceClick}
  />);
  const button = component.find('.ProduceBtn').first();
  button.simulate('click');

  setTimeout(() => {
    expect(handleProduceClick.mock.calls.length).toBe(1);
    expect(handleProduceClick.mock.calls[0][0]).toBe('Produce');
    done();
  });
});
