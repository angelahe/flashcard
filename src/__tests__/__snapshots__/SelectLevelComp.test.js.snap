// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of UI renders consistently 1`] = `
<div
  className="ChooseLevelComp"
>
  <div
    className="LineContainer"
  >
    <button
      className="LevelImage ReviewBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="review"
        className="DeckImg"
        src="eye_FFFFFF.png"
      />
    </button>
  </div>
  <div
    className="LineContainer"
  >
    <button
      className="LevelImage RecognizeBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="recognize"
        className="DeckImg"
        src="read_FFFFFF.png"
      />
    </button>
  </div>
  <div
    className="LineContainer"
  >
    <button
      className="LevelImage ProduceBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="produce"
        className="DeckImg"
        src="write_FFFFFF.png"
      />
    </button>
  </div>
</div>
`;
